core/src/subgraph/context/instance.rs

```diff
@@ -215,10 +220,17 @@ where
         self.causality_region_seq.next_val()
     }
 
-    #[cfg(debug_assertions)]
     pub fn hosts(&self) -> &[Arc<T::Host>] {
         &self.hosts.hosts()
     }
+
+    pub fn static_data_sources(&self) -> Arc<Vec<DataSource<C>>> {
```

Comment:
Why?

Reply:
@mangas I dont understand, are you suggesting making the field public instead of exposing it in a method?

Reply:
I don't think we should be adding getter methods unless they are absolutely needed. I'm also not too happy about introducing "static_data_sources" as a term, historically we have referred to these as "data_sources" and "dynamic_data_sources" if we need this distinction then perhaps it should be done with types

Reply:
makes sense, so better making it `pub` than exposing it in a method, got it.

Also i think yes we can rename it to `data_sources` instead, This is true for `manifest.data_sources` where only "static" datasources where stored. 

We can keep that naming, i was trying to make a clear distinction because, Although the `SubgraphManifest.data_sources` was meant to be just the "static" datasources, it used to carry both static and dynamic datasources when its mutated to add dynamic sources in `build_subgraph_runner`.